### PR TITLE
Remove second argument to .save()

### PR DESCRIPTION
**Pull request by particlebanana**
> This is a breaking change and will require us to go ahead and push out a smaller version of `0.11`.

This may be controversial but I think it's the right thing to do. The current behavior of `.save()` is awful as documented in #441 #1091 #1037 #954 and probably more. Right now it populates all the attributes which can easily overflow memory. There is a flag to mitigate some of this but it shouldn't happen in the first place.

Trying to determine what to populate based on what attributes are currently set in the model sounds nasty as well. Removing this attribute makes the API cleaner and more predictable as well as a tad bit faster because it has do at minimum one less query.

I want to continue removing pieces related to model serialization in future releases as well so nested/graph operations are probably next on the chopping block.

So to summarize this changes the function signature of the callback passed to `.save()` from `.save(function(err, updatedRecord) {})` to `.save(function(err) {})`.

### CODE REVIEW

**Comment by devinivy** [comment]
Does this affect `save()`'s promised interface in the way one would expect?

**Comment by particlebanana** [comment]
@devinivy it changes the promise resolver from:

```javascript
if (deferred) {		
  deferred.resolve(data);		
}
```

```javascript
if (deferred) {		
  deferred.resolve();		
}
```

I don't use promises does this cause some unexpected issue?

**Comment by devinivy** [comment]
That seems right!  Just makin' sure, since it seems like the type of thing that could get buggered.  Still thinking about two things,

1. Can we reset the values on the saved instance with the new values from the db?  Imagine saving an object, defaults being applied to an attribute, the update occurring successfully, then the model instance not reflecting the defaults that were applied.
2. Can we make an additional breaking change to pass a legit `Error` rather than an array of objects representing failed transactions?

**Comment by particlebanana** [comment]
@devinivy I think both of those requests are valid. I'd like to try and get this out today. What should the `Error` look like you think?

**Comment by devinivy** [comment]
Cool!  For the error maybe something simple like,
```js
var err = new WLError('Some associations could not be added or destroyed during save().');
err.failedTransactions = failedTransactions; // The array of failed transactions
```

**Comment by pauln** [comment]
Is there any reason that the population options were removed completely, rather than simply changing the default (to `false`, so that nothing is populated by default but it's possible to specify keys to populate - or set it to `true` if you really want to populate all associations)?  It doesn't seem to be possible to `populate()` after a `save()` - I get `myObj.save(...).populate is not a function` - so with the population options gone, I can't see a way to populate the associations at all.  This means that although I had an association populated before saving, once I've saved, it's been flattened with no way to populate it again - which seems like a strange side-effect of saving.

If I'm missing something, please enlighten me - but otherwise, can we please either have the population options back (with appropriate defaults) or get support for `.save().populate()` implemented?
